### ISSUE TITLE
on-remove and on-select should fire AFTER addition/removal of the item.

### ISSUE THREAD

**Issue by gotenxds** [angular-ui/ui-select#463]
I have a list of entities, each entity has a list of items, those items are the ones I'm displaying to the user in multiple mode.

When the user removes or adds and item to the list the on-action event fires, but it fires before the removal\addition of the item from the list, so I can not send my entity for update in the DB.

Side note: This will also compliment the way ui-grid works on edit.
Example to what I mean:

html:
```html
<ui-select multiple ng-model="exemption.selected.dutyTypes" theme="select2" on-remove="update" >
```

script (pseudo code):
```javascript
$scope.update = function(){
     $http.put("/exemptions/", exemption.selected); // cant do this because the item is not updated yet!!!
}
```

**Comment by ilanbiala** [comment]
I think this is fixed now, but it would be great if someone else could confirm.

**Comment by brianfeister** [comment]
Resolved by #501

**Comment by saurabhsoni14** [comment]
how to get only ID and post in database
